torch/fx/interpreter.py

```diff
@@ -170,6 +170,7 @@ def boxed_run(self, args_list):
         calling convention, where you pass a list of arguments, which will be cleared
         by the interpreter.  This ensures that input tensors are promptly deallocated.
         """
+        assert len(args_list) == len(self.graph.find_nodes(op="placeholder"))
```

Comment:
One risk is that it's possible for people to (incidentally) end up using the `fx.Interpreter` to run their code in the hotpath (e.g. if they are using export, and running the exported code without compilation).

Just to reduce risk: if you want to keep the assert in, can you put it in a separate PR?